### ISSUE TITLE
Don’t warn about unset in loop on object-like array

### ISSUE THREAD

**Issue by muglug** [vimeo/psalm#1628]
```php
/**
 * @param string[] $arr
 */
function foo(array $arr) : void {
  $dict = ['a' => 1, 'b' => 2, 'c' => 3];

  foreach ($arr as $v) {
    unset($dict[$v]);
  }

  if (count($dict)) {}
}
```

Expected: No issue
Actual: `RedundantCondition`: https://psalm.dev/r/a21a6d112f
